src/_pytest/recwarn.py

```diff
@@ -42,7 +55,32 @@ def deprecated_call(func=None, *args, **kwargs):
     return warns((DeprecationWarning, PendingDeprecationWarning), *args, **kwargs)
 
 
-def warns(expected_warning, *args, match=None, **kwargs):
+@overload
+def warns(
+    expected_warning: Union["Type[Warning]", Tuple["Type[Warning]", ...]],
```

Comment:
we see this pattern of union of x, tuple of x so often, i wonder if it should get a name to spell more comprehensible

Reply:
We can define a generic type alias

```py
TOrTupleOfT = Union[T, Tuple[T, ...]]
```

But I'm not sure that the indirection is worth it, at least yet.

Reply:
I'm marking as resolved -- let me know if you think it's worth doing.